tests/test_resolve_recorded_replay_window.py

```
@@ -0,0 +1,128 @@
+class ResolveRecordedReplayWindowTests(unittest.TestCase):
+    def run_script(self, recording_rows, dryrun_payload):
+        with tempfile.TemporaryDirectory() as tmp:
+            tmp_path = Path(tmp)
+            recording = tmp_path / "recording.ndjson"
+            dryrun = tmp_path / "dryrun.json"
+            output = tmp_path / "window.json"
+            env = tmp_path / "window.env"
+            write_ndjson(recording, recording_rows)
+            dryrun.write_text(json.dumps(dryrun_payload), encoding="utf-8")
+
+            subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)